sdk/keyvault/azure-security-keyvault-secrets/src/test/java/com/azure/security/keyvault/secrets/SecretAsyncClientTest.java

```diff
@@ -310,6 +310,17 @@ public void recoverDeletedSecretNotFound(HttpClient httpClient, SecretServiceVer
             .verifyErrorSatisfies(ex -> assertRestException(ex, ResourceNotFoundException.class, HttpURLConnection.HTTP_NOT_FOUND));
     }
 
+    /**
+     * Tests that an attempt to recover a non existing deleted secret throws an error on a soft-delete enabled vault.
+     */
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("getTestParameters")
+    public void recoverDeletedSecretNotFoundWithPoller(HttpClient httpClient, SecretServiceVersion serviceVersion) {
```

Comment:
The difference between this method and the one above is that this one accepts a polling duration, while both use a `PollerFlux` under the hood.
```suggestion
    public void recoverDeletedSecretNotFoundWithPollingDuration(HttpClient httpClient, SecretServiceVersion serviceVersion) {
```